src/main/java/com/github/pedrovgs/androidwifiadb/view/AndroidDevices.java

```diff
@@ -0,0 +1,170 @@
+            @Override
+            public void run() {
+                cardLayoutDevices.createAndShowGUI();
+            }
+        });
+    }
+
+    /**
+     * Monitors connected devices state and triggers UI updates.
+     */
+    private void monitorDevices() {
+        ApplicationManager.getApplication().executeOnPooledThread(new Runnable() {
+            public void run() {
+                while (true) {
+                    boolean refreshRequired = androidWifiADB.refreshDevicesList();
+                    if (refreshRequired) {
+                        updateUi();
+                    }
+
+                    try {
+                        Thread.sleep(INTERVAL_REFRESH_DEVICES);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public void showNoConnectedDevicesNotification() {
+        showNotification(ANDROID_WIFI_ADB_TITLE,
+                "There are no devices connected. Review your USB connection and try again. ",
+                NotificationType.INFORMATION);
+    }
+
+    @Override
+    public void showConnectedDeviceNotification(Device device) {
+        showNotification(ANDROID_WIFI_ADB_TITLE, "Device '" + device.getName() + "' connected.",
+                NotificationType.INFORMATION);
+    }
+
+    @Override
+    public void showDisconnectedDeviceNotification(Device device) {
+        showNotification(ANDROID_WIFI_ADB_TITLE, "Device '" + device.getName() + "' disconnected.",
+                NotificationType.INFORMATION);
+    }
+
+    @Override
+    public void showErrorConnectingDeviceNotification(Device device) {
+        showNotification(ANDROID_WIFI_ADB_TITLE,
+                "Unable to connect device '" + device.getName() + "'. Review your WiFi connection.",
+                NotificationType.INFORMATION);
+    }
+
+    @Override
+    public void showErrorDisconnectingDeviceNotification(Device device) {
+        showNotification(ANDROID_WIFI_ADB_TITLE,
+                "Unable to disconnect device '" + device.getName() + "'. Review your WiFi connection.",
+                NotificationType.INFORMATION);
+    }
+
+    @Override
+    public void showADBNotInstalledNotification() {
+        showNotification(ANDROID_WIFI_ADB_TITLE,
+                "'adb' command not found. Review your Android SDK installation.", NotificationType.ERROR);
+    }
+
+    private void showNotification(final String title, final String message,
+                                  final NotificationType type) {
+        ApplicationManager.getApplication().invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                Notification notification =
+                        NOTIFICATION_GROUP.createNotification(title, message, type, null);
+                Notifications.Bus.notify(notification);
+            }
+        });
+    }
```

Comment:
If I'm not wrong, this method is duplicated. We should extract this functionallity to a ``NotificationUtils`` class.